### PR TITLE
add support for memberof

### PR DESCRIPTION
**Pull request by mrow4a**
Cornelius asked for this @cdamken @felixboehm Please review and tell me if this makes sense etc, this does some changes.

### CODE REVIEW

**Comment by mrow4a** [comment]
I dont actualy remember what was that, but hope you can use it Cornelius.

**Comment by dercorn** [comment]
@mrow4a Thank you very much. I'll have a look at it.

**Comment by cdamken** [comment]
Looks fine, my only comment is why do we use the base:
``` 
-LDAP_BASE_DN=dc=owncloud,dc=com
``` 
if there is a test environment we could change it, that way it wont us if in the future we have the production LDAP from the company.

**Comment by mrow4a** [comment]
I mean it is just development branch, you can get the branch localy, change it, and merge this PR as required.

**Comment by mrow4a** [comment]
Do we merge it? Is it useful?

**Comment by PVince81** [comment]
I'll try this branch next time I need to setup LDAP. So far the code changes look fine.

**Comment by PVince81** [comment]
:-1:

```
± % ./start_ldap.sh 
./start_ldap.sh: line 52: syntax error: unexpected end of file
```

**Comment by mrow4a** [comment]
Dont know why, this files has 51 files, no no idea why it has an error on 52th

**Comment by mrow4a** [comment]
This branch hangs there since Aug 2, 2016, so you know..

**Comment by DeepDiver1975** [comment]
> Dont know why, this file has 51 lines, no no idea why it has an error on 52th

missing new line?

**Comment by mrow4a** [comment]
I will rethink this script tomorrow and update this script if needed.

**Comment by mrow4a** [comment]
New commit amend allows using memberof testing ldap, and also creates users within multiple groups (previously you could create groups, or users, now mixed -> batchCreateUsersInGroups.php)

@PVince81 @butonic

File: ldap-testing/batchCreateUsersInGroups.php [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+$entry['ou'] = 'Zombies';
+$b = ldap_add($cr, $ouDN, $entry);
+
+$names = json_decode(file_get_contents('names.json'), true);
+$cfn = count($names['fns']) - 1;
+$csn = count($names['sns']) - 1;
+$zombie_counter=0;
+
+#CREATE KING
+$uid= 'zombieKing';
```
Comment (PVince81) [inline ldap-testing/batchCreateUsersInGroups.php]: 😆

**Comment by PVince81** [comment]
By default it seems the armies do not have any members, only the zombieKing (memberUid). Is this on purpose ?

File: ldap-testing/config.php [hunk 2]
```diff
@@ -1,7 +1,23 @@
 <?php
+  // Basic params
   $bdn = 'dc=owncloud,dc=com';
   $adn = 'cn=admin,dc=owncloud,dc=com';
   $apwd = 'admin';
   $host = 'localhost';
   $port = 389;
 
+  // Define amount of users in groups
+  $amount_users_in_groups = 0;
```
Comment (mrow4a) [inline ldap-testing/config.php]: Yes, @PVince81 by default they have no members, only zombieKing. Should we make it more transparent? (README)?

Comment (PVince81) [reply ldap-testing/config.php]: can you set the default to some non-zero value maybe ?

Comment (mrow4a) [reply ldap-testing/config.php]: makes sense, otherwise it is just confusing

**Comment by mrow4a** [comment]
done

**Comment by PVince81** [comment]
:+1:

**Pull request merged**